pkg/operator/controllers/clusteroperatoraro/clusteroperatoraro_controller.go

```diff
@@ -74,44 +77,28 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 		return reconcile.Result{}, err
 	}
 
-	return reconcile.Result{}, r.setClusterOperatorStatus(ctx, co)
+	return reconcile.Result{}, r.setClusterOperatorStatus(ctx, co, instance)
 }
 
-func (r *Reconciler) setClusterOperatorStatus(ctx context.Context, co *configv1.ClusterOperator) error {
-	// TODO: Replace with a real conditions based on aro operator conditions
-	currentTime := metav1.Now()
-	conditions := []configv1.ClusterOperatorStatusCondition{
-		{
-			Type:               configv1.OperatorAvailable,
-			Status:             configv1.ConditionTrue,
-			LastTransitionTime: currentTime,
-			Reason:             reasonAsExpected,
-		},
-		{
-			Type:               configv1.OperatorProgressing,
-			Status:             configv1.ConditionFalse,
-			LastTransitionTime: currentTime,
-			Reason:             reasonAsExpected,
-		},
-		{
-			Type:               configv1.OperatorDegraded,
-			Status:             configv1.ConditionFalse,
-			LastTransitionTime: currentTime,
-			Reason:             reasonAsExpected,
-		},
-		{
-			Type:               configv1.OperatorUpgradeable,
-			Status:             configv1.ConditionTrue,
-			LastTransitionTime: currentTime,
-			Reason:             reasonAsExpected,
-		},
-	}
+func (r *Reconciler) setClusterOperatorStatus(ctx context.Context, originalClusterOperatorObj *configv1.ClusterOperator, cluster *arov1alpha1.Cluster) error {
+	clusterOperatorObj := originalClusterOperatorObj.DeepCopy()
+
+	v1helpers.SetStatusCondition(&clusterOperatorObj.Status.Conditions, status.UnionClusterCondition("Available", operatorv1.ConditionTrue, nil, cluster.Status.Conditions...))
+	v1helpers.SetStatusCondition(&clusterOperatorObj.Status.Conditions, status.UnionClusterCondition("Progressing", operatorv1.ConditionFalse, nil, cluster.Status.Conditions...))
+
+	// We always set the Degraded status to false, as the operator being in Degraded state will prevent cluster upgrade.
+	v1helpers.SetStatusCondition(&clusterOperatorObj.Status.Conditions, configv1.ClusterOperatorStatusCondition{
+		Type:               configv1.OperatorDegraded,
+		Status:             configv1.ConditionFalse,
+		LastTransitionTime: metav1.Now(),
+		Reason:             reasonAsExpected,
+	})
 
-	for _, c := range conditions {
-		v1helpers.SetStatusCondition(&co.Status.Conditions, c)
+	if equality.Semantic.DeepEqual(clusterOperatorObj.Status.Conditions, originalClusterOperatorObj.Status.Conditions) {
```

Comment:
Won't this comparison always return false due to the LastTransitionTime being modified on line 93?

Reply:
The `v1helpers.SetStatusCondition` function from openshift/library-go will do its own diff on the "existing" and "new" status conditions, and will only update the tiemstamp to the "new" condition's timestamp timestamp if the actual status has changed:

https://github.com/openshift/library-go/blob/f3277c7720118c0b91bda69112f46a4928bf22b8/pkg/config/clusteroperator/v1helpers/status.go#L29